# random comments
# it could be noted that some 'marker for a certain cell type' is not agreed by different datasets. Personal idea is that when to define cell types, prioritize genes with known biological functions, rather than those computationally identified.

#================================================================#
# use standard naming system, store in variable
#================================================================#

# general tumor infiltraing T cell types
cd8.Tn <- "CD8+ T naive"
cd8.Teff <- "CD8+ T effector"
cd8.Tem <- "CD8+ T effector memory"
cd8.Trm <- "CD8+ T resident memory"
cd8.Tex <- "CD8+ T exhausted"
cd4.Treg <- "CD4+ T regulatory"
cd4.Tn <- "CD4+ T naive"
cd4.Tex <- "CD4+ T exhausted"
cd4.Th17 <- "CD4+ Th17" # use IL23R
cd4.Tem <- "CD4+ T effector memory"
cd4.others <- "CD4+ others"

#================================================================#
# useful markers
#================================================================#

conventional.T.cell.markers <- c(
  'CD3D',
  'CD4',
  'CD8A',
  'FOXP3'
)

effector.markers <- c(
  
  'GZMA', # granzyme A has minimal cytotoxicty [39]
  'GZMB', # GZMB is the most powerful [39]
  'GZMK', # granzyme expression during the differentiation process of memory CD8+ T cells might be as follows: GrA+/GrB–/GrK+ ? GrA+/GrB+/GrK+ ? GrA+/GrB+/GrK– [12] ranked in early effector memory subset [3, 13]
  'GZMH', # GZMH & GZMK may expain the minimal cytotoxcitiy effect in abscence of both A&B [39]
  "GZMM", # expressed less in Tex compared to Teff [1~3] reduced expression in Tex compared to Teff [1~3]
  
  'GNLY', # as name 'granulysin' suggests, exists within GZMs vesicles and causes lysis [38]. seems mostly to kill bacteria and low toxicity to mammalian cells [39]
  
  'PRF1', # contained within vesicles [39] regulated by RUNX3 & EOMES [39] 
  "UNC13D", # claimed to regulate prf vesicle trafficking & fusion [39], expressed low & not very differentially between Teff & Tex [1~3]
  "STXBP2", # claimed to regulate prf vesicle trafficking & fusion [39], expressed low & not very differentially between Teff & Tex [1~3]
  "STX11", # claimed to regulate prf vesicle trafficking & fusion [39], expressed low & not very differentially between Teff & Tex [1~3]
  "DOCK8", # claimed to regulate prf vesicle trafficking & fusion [39], expressed in almost all T cells [1~3]
  "LAMP1", # claimed to be redundant in cytotoxicity [39] mentioned to translocate to plasma membrane [36]
  
  'IFNG', 
  
  "CX3CR1", # seem to be a good marker. associated with effector T cell subset although function not very clear[20] seems only used by zhanglab, not that good in [22]
  
  "NKG7", # transmembrane protein localized on vesicles containing cytotoxic molecules. relocalizing to plasma membrane following membrane fusion of vesicles with plasma membrane, when T/NK cells recognize target cells and form a immunological synapse. reduced LAMP1 on plasma membrane in NKG7 difficiency suggests a role in vesicle trafficing. [36, 37] expressed in a all but naive CD8 T cells. [1~3] so may not contribute to T cell exhuastion. claimed roles in CD4 T cells [36~37]
  
  "FCGR3A", # also known as CD16, originally functions to bind the Fc fragment of antibody to enable clearance [59] expressed exclusively in Teff, might be interesting [1~3]
  "FCGR3B", # almost the same as FCGR3A but significantly less amount [1~3]
  
  "FGFBP2", # The encoded protein is a serum protein that is selectively secreted by cytotoxic lymphocytes and may be involved in cytotoxic lymphocyte-mediated immunity. [60] exclusively expressed by Teff [1~3]
  
  
)

naive.markers <- c(
  "LEF1",
  "CCR7",
  "SELL",
  "TCF7",
  
  "MAL", # correlates surprisingly well (tSNE gene expression) with CCR7 [1~3], functions in immunology not completely understood but well summarized[7]. worth further looking at if have time. Biomarker in treatment response a story? 
  "IL7R", # see other IL7R
)

memory.markers <- c(
  
  # effector memory
  "CXCR4", # expressed in almost all T cells but slightly higer in Tem. ranked as marker of Tem[3]. function similar to CXCR3, expressed in a variety of immune cells [15]
  "CXCR3", # ligands: CXCL9, CXCL10 and CXCL11 are commonly produced by local cells in inflammatory lesion, suggesting that CXCR3 and its chemokines participate in the recruitment of inflammatory cells[14]. marker of Tem[14, 3].
  "CD44", # cell adhesion molecule, expressed "in many cell types" [8]. expressed on almost all T cells [3]. indicated as 'effector-memory T cell marker' [8, 9]
  "CCR5", # breast cancer expression & target, involved in HIV. T cell function not well established [10]. expression related to but not well-distinguish Tem [2, 3]. Suggested as Tem marker [11]. likely that CCR5 plays a role in inflammatory responses to infection, though its exact role in normal immune function is unclear[10]
  "GZMK", # see above
)

resident.markers <-c(

  "CD69", # [18] The general idea is straightforward in that CD69 & ITGAE are related to cell-adhesion molecules. Binding parteners of CD69 & CD103 is expressed in tissue (epithelial, etc? check [18])
  "ITGAE", # also CD103 [18]
  "ZNF683", # related to Trm [13, 19]
)

exhaustion.markers <- c(

  "PDCD1",
  "TIGIT",
  "LAYN"
  
)

inflamatory.markers <- c(

  "CCL4", # interacts with CCR5. chemoattractant for natural killer cells, monocytes and a variety of other immune cells. produced by: neutrophils, monocytes, B cells, T cells, fibroblasts, endothelial cells, and epithelial cells. [16]
)

homing.related <- c(

  "GPR183", # ranked for Tcm [3] but not very distinguising. functions not very clear, seem to relate to T/B homing within lymph nodes (generally related to 'central') but well summarized [17]
)

follicular.markers <- c(  # Distinguishing features of TFH cells are the expression of CXCR5, PD-1, SAP (SH2D1A), IL-21, and ICOS, among other molecules, and the absence of Blimp-1 ( prdm1). TFH cells are important for the for- mation of germinal centers [23]
  "CXCR5", # [23]
  "BCL6", # TFH cells de- pend on expression of the master regulator transcription factor Bcl6. [23]
)

t.cell.transcription.factors <- c(  
  
  # do note transcription factors are often considered undruggable, couldn't be drug targets.
  
  # it seems easy to discover a lot of these things using 'correlation in transcriptome', starting from some keynote genes such as TOX, TCF7
  # most of these are summarized nicely in box2 [5]
  
  # below are 'established'
  "TCF7", # expressed exclusively in 'progenitor exhausted' [24] this conclusion seem to be supported by many many studies and TCF7 subset associates with improved prognosis. [25~27]
  "TOX", # highly expressed in dysfuntional T cells in tumor and exhaustive T cells in chronic viral infection. high expression of TOX correlated with high expression of inhibitory receptors and low expression of TCF7[28] validated in [6]. also expressed in Treg & CD4 Tex [1, 6] plays in the elegant control of long-time T cell activation (exhaustion), have suggested functions in TCF1 cells [30] dispensable for Tem but essential for Tex [31]
  
  # below are less validated
  "NFATC2", # encodes NFAT1 [29], NFAT1 claimed to regulate TOX [28]. NFATs expression not that differential [1]
  "SETBP1", # identified in [31] to correlate with TOX. acts like a DNA-binding protein [32] express but not significantly in Tex & Treg [1, 3] (low detection reasonable, given being low-expression DNA binding proteins). Might be upstream regulators of TOX, etc. seems not very important.
  "IKZF2",  # identified in [31] to correlate with TOX. zinc finger as name suggests, involved in hematopoietic development [33] 
  "NR4A2", # claimed to be expressed in Tex [34], validated but not that significantly or consistently across datasets [1~3]
  "MAF", # claimed to be expressed in Tex [34], validated but not that significantly or consistently across datasets [1~3]
  "EOMES", # validated [13] for the claim: as tumor progressed, Eomes expression was elevated in tumor-infiltrating CD8+ T cells, especially in PD-1+Tim-3+ exhausted CD8+ T cells. Complete loss of Eomes in T cells resulted in impaired development of anti-tumor CTLs, whereas deletion of one allele of Eomes in T cells decreased development of exhausted CD8+ T cells, which offered better tumor control. [35]
  "RUNX3", # regulate PRF1 expression [40], expressed in Teff & Tex in [1~3]
  "BATF", # claimed to be upregulated by PD1 activdation [46] seem to express in all T cells but significantly higher in Treg. [46]
)

costimulatory.receptors <- c(
  "CD28", # TARGET # amplify TCR signaling, induce proliferation & IL2 production [41] expressed in GZMK T cells, T naive but not Teff, Tex [1~3] CTLA4 is upregulated following T cell activation, with CD28 subsequently downregulated. [41, 42] the question is, will Tex work without CD28? CD28 is dephosphorelayed by SHP2 recruited by PDL1-PD1 activation, suggests CD28 is the primary target of PD1 activation. [43] failed in clinical trial, perhaps due to super efficiency [44]
)

coinhibitory.receptors <-c(
  "LAG3", # TARGET # starts from Teff, climaxed at Tex [1~3], interacts with MHC-II [41]  MHCII binding may distinguish it from other targets [45, 51]
  "CTLA4", # TARGET # sequestering target receptors or ligands[47]
  "PDCD1", # TARGET # recruiting SHP2 to attenuate TCR signaling [48] rapidly upregulated following T cell activation []
  "TIM3", # TARGET # involved in synapse formation, recruited by PD1 [49]
  "KLRG1", # expressed less in Tex compared to Teff [1~3] immune checkpoint with less obvious functions than PD1, etc. considered as a 'senescent marker' expressed in effector, etc [21]
)

costimulatory.receptors <-c(
  "TNFRSF4", # also known as OX-40, expresses in TNFRSF9+ Treg, but not CD8 Tex. express in CD4 Tex [2]
  "TNFRSF9", # also known as 4-1BB,  very interesting to express only in terminal Tex & Treg, why would 'stimulatory' be on Tex? [1~3] express on antigen-experienced Treg [2]
  ""
)

inteferons <- c(

  "IFNG", # links to JAK1/JAK2 signaling, IFNG upregulates typeI antigen presentation & increases PDL1 expression [72] direct anti-proliferative effect [need reference]
  
  
)

interleukins <- c(
  
  "IL2RG", # The common gamma chain (γc) (or CD132), also known as interleukin-2 receptor subunit gamma or IL-2RG, is a cytokine receptor sub-unit that is common to the receptor complexes for at least six different interleukin receptors: IL-2, IL-4, IL-7, IL-9, IL-15 and interleukin-21 receptor. [61] expressed on all TILs [1]
  
  "IL2", # TODO, express in Treg, responsible for t cell maintainence [62] # tested in combination with PDL1 blockade and worked (but IL2RA expressed in Treg, IL15 chosen by genentech as 'better-in-class')[57] 
  
  ""
  "IL7R", # in pan-T ranked as a marker. However, expressed in almost all cd8&4 except exhausted [1~3]. involved in VDJ recombination [4]. suggested in Melanoma & BCC, could be skin-cancer enriched? [5] but not validated in [6]. Don't suggest to use when assigning clusters. Could use in retrospection. In fact, almost useless in defining Naive.
  
  
  "IL10" # inhibitory cytokine. promotes T cell exhaustion [52] # not expressed by any TIl [1~3] produced by monocytes [54~55], validated by [53] in specific macrophage subsets. coblockade with PD1 tested [56]
  "IL10RA", # receptor # seem to express low but in all TIL [1~3] 
  "IL10RB", # receptor # seem to express low but in all TIL [1~3] IL10RB expression seems to be subsets of IL10RA [1, 53], why?
  
  "IL15", # genentech target # the expression of IL15 doesn't seem robust [53]
  "IL15RA", # expression low [1~3]
  
  "IL17A", # the '17' that defines Th17 [65] expressed exclusively by Th17 [3], 
  "IL17F", # very similar to IL17A, expressed by Th17 [65]
  "IL17B", # no expression in any TIl [3]
  "IL17C", # no expression in any TIl [3]
  "IL17D", # no expression in any TIl [3]
  "IL17RA", # seem to express in all TIL in a very sparse fashion [3]
  
  "IL23A", # driving factors of Th17 development, source not very clear [65]
  "IL23R", # almost exclusively on Th17 [3]
  
  "IL33", # target, via IL33 antagonist[67] # recruits Treg to the TME[66]. source seems to be endothelial but not cancer [68, 69] PD1 & CTLA4 treatment upregualtes ST2 in tumor & promotes CD103 CD8 Trm, this seems not solid. [69]
  "IL1RL1" # target, IL1-receptor like 1, receptor for IL33. [66], expressed exclusively by Treg & Tfr [3] in a subset of Treg [2] but not in [1]
)

chemokines <- c(
  
  "CCR7", # well-establishes to Lymph-node homing [63]
  "CCL19", # CCR7 ligands expressed by HEV epithelial [63], not expressed by T cells [3]
  "CCL21", # CCR7 ligands expressed by HEV epithelial [63], not expressed by T cells [3]
  
  "CXCL13", # receptor is CXCR5, CXCL13 is a attractant for B & Tfh to promote TLS formation. CD8 T cell acquire increased expression of CXCL13 along the dysfunctional lineage. Thus Tex should not be considered as inert but with designed functions [5] Expressed in Tex [1~3] together with ITGAE [58]. 
)



homing.related <- c(

  "S1PR1", # S1P (sphingosine 1-phosphate) expresses abundantly in blood & lymph. Binding of S1P transitently attenuates S1PR1 [63] enabling T cell migration to lymph node. express in CD8 & CD4 Teff & T naive [1~3] Personally consider it is a marker indicating 'should be in blood', corroborated by STARTRAC [3]

  "ANXA1", # encodes a membrane-localized protein that binds phospholipids. related to anti-inflammatory functions not well-defined functions [64], doesn't seem like a good marker, express in almost all cell [1~3] though claimed as marker of Blood-Tcm in [3]
  "ANXA2", # even worse than ANXA1.
)

other.genes.DNA.binding <- c(
)

#================================================================#
# genes incolved in mutation & resistance to immunotherapy
#================================================================#

genes.mutation.immunotherapy <- c(
  
  "IFNGR1", # see below JAK
  
  "JAK1", # loss of function or truncating mutation leads to insensitivity to INFG related pathways, i.e. INFG for antiproliferative functions in cancer cells. [70] think JAK is involved in both tumor & immune cells?
  "JAK2", # as JAK1 [70]
  
  "B2M", # truncating mutation leads to loss of MHC-I [70] common component pf HLA isotypes. [] point mutation, deletions or loss of hyterozygosity observed in non-responders(LOH), LOH more 3-fold in non-responder (30% to 10%) than responder, loss of two copies only in non-responder [73]
  
  
)

histology.markers <- c(
  
  "S100", # tumor marker for melanoma, not necessarily for other cancer type [71]
  
)

# references:
# 1. Landscape of infiltrating T cells in liver cancer revealed by single-cell sequencing, Cell, 2017
# 2. Global characterization of T cells in non-small-cell lung cancer by single-cell sequencing, Nature Medicine, 2018
# 3. Lineage tracking reveals dynamic relationships of T cells in colorectal cancer, Nature, 2018
# 4. https://en.wikipedia.org/wiki/Interleukin-7_receptor
# 5. CD8+ T cell states in human cancer: insights from single-cell analysis, Nature review cancer, 2020
# 6. Clonal replacement of tumor-specific T cells following PD-1 blockade, Nature medicine, 2019
# 7. https://www.wikigenes.org/e/gene/e/4118.html
# 8. https://en.wikipedia.org/wiki/CD44
# 9. Memory T Cell Subsets, Migration Patterns, and Tissue Residence, annual review immunology, 2013
# 10. https://en.wikipedia.org/wiki/CCR5
# 11. Three Memory Subsets of Human CD8 T Cells Differently Expressing Three Cytolytic Effector Molecules, 
# 12. Differential expression of human granzymes A, B, and K in natural killer cells and during CD8+ T cell differentiation in peripheral blood
# 13. pan-T zhanglab
# 14. https://en.wikipedia.org/wiki/CXCR3
# 15. https://en.wikipedia.org/wiki/CXCR4
# 16. https://en.wikipedia.org/wiki/CCL4
# 17. https://www.genecards.org/cgi-bin/carddisp.pl?gene=GPR183
# 18. Memory T Cell Subsets, Migration Patterns, and Tissue Residence, Annual review
# 19. https://www.genecards.org/cgi-bin/carddisp.pl?gene=ZNF683
# 20. (2015). Functional classification of memory CD8(+) T cells by CX3CR1 expression. Nat. Commun.
# 21. https://en.wikipedia.org/wiki/KLRG1
# 22. Clonal replacement of tumor-specific T cells following PD-1 blockade, Nat Med 2019
# 23. Follicular Helper CD4 T Cells (TFH), Annual Reviews immunology, 2011
# 24. Subsets of exhausted CD8+ T cells differentially mediate tumor control and respond to checkpoint blockade, Nature immunology 2019
# 25. Defining T Cell States Associated with Response to Checkpoint Immunotherapy in Melanoma, Cell, 2018
# 26. Defining CD8+ T cells that provide the proliferative burst after PD-1 therapy, Nature, 2016
# 27. Intratumoral Tcf1+PD-1+CD8+ T Cells with Stem-like Properties Promote Tumor Control in Response to Vaccination and Checkpoint Blockade Immunotherapy, Immunity, 2019
# 28. TOX is a critical regulator of tumour-specific T cell differentiation, Nature, 2019
# 29. https://www.cellsignal.com/products/primary-antibodies/nfat1-antibody/4389
# 30. TOX reinforces the phenotype and longevity of exhausted T cells in chronic viral infection, Nature, 2019
# 31. TOX transcript+ionally and epigenetically programs CD8 T cell exhaustion, Nature, 2019
# 32. https://en.wikipedia.org/wiki/SETBP1
# 33. https://en.wikipedia.org/wiki/IKZF2
# 34. Molecular profiling of CD8 T cells in autochthonous melanoma identifies Maf as driver of exhaustion, EMBO 2015
# 35. High Levels of Eomes Promote Exhaustion of Anti-tumor CD8+ T Cells, Trends in immunology
# 36. NKG7 makes a better killer, Nature immunology, 2019
# 37. The NK cell granule protein NKG7 regulates cytotoxic granule exocytosis and inflammation, Nature immunology, 2019
# 38. https://en.wikipedia.org/wiki/GNLY
# 39. Perforin and granzymes- function, dysfunction and human pathology, Nature review immunology, 2015
# 40. Runx3 and T-box proteins cooperate to establish the transcriptional program of effector CTLs. J. Exp. Med. 206, 51–59 (2009).
# 41. Mol. Cell. Biol. 7, 4472–4481 (1987). This paper identified CD28 as a co‐stimulator that could amplify TCR signalling to induce proliferation and IL‐2 production
# 42. CD28 and CTLA-4 coreceptor expression and signal transduction. Immunol. Rev. 229, 12–26 (2009).
# 43. T cell costimulatory receptor CD28 is a primary target for PD-1–mediated inhibition, Science 2017
# 44. The storm has cleared- lessons from the CD28 superagonist TGN1412 trial, Nature reviews immunology, 2012
# 45. https://blog.crownbio.com/lag3-a-promising-new-cancer-immunotherapy-target
# 46. Transcriptional analysis of HIV-specific CD8+ T cells shows that PD-1 inhibits T cell function by upregulating BATF, Nature medicine 2010
# 47. B7-1 and B7-2 selectively recruit CTLA-4 and CD28 to the immunological synapse. Immunity 21, 401–413 (2004).
# 48. Programmed cell death 1 forms negative costimulatory microclusters that directly inhibit T cell receptor signaling by recruiting phosphatase SHP2. J. Exp. Med. 209, 1201–1217 (2012).
# 49. T cell Ig and mucin domain- containing protein 3 is recruited to the immune synapse, disrupts stable synapse formation, and associates with receptor phosphatases. J. Immunol. 192, 782–791 (2014).
# 50. Molecular and cellular insights into T cell exhaustion, Nature reviews immunology 2015
# 51. Clinical blockade of PD1 and LAG3 — potential mechanisms of action, Nature reviews immunology, 2014
# 52. Ejrnaes, M. et al. Resolution of a chronic viral infection after interleukin-10 receptor blockade. J. Exp. Med. 203, 2461–2472 (2006).
# 53. Landscape and Dynamics of Single Immune Cells in Hepatocellular Carcinoma, Cell, 2019
# 54. Infected CD8α– dendritic cells are the predominant source of IL-10 during establishment of persistent viral infection. Proc. Natl Acad. Sci. USA 109, 14116–14121 (2012).
# 55. Programmed death-1-induced interleukin-10 production by monocytes impairs CD4+ T cell activation during HIV infection. Nat. Med. 16, 452–459 (2010).
# 56. IL-10 and PD-L1 operate through distinct pathways to suppress T-cell activity during persistent viral infection, PNAS, 2008
# 57. PD-L1 blockade synergizes with IL-2 therapy in reinvigorating exhausted T cells. J. Clin. Invest. 123, 2604–2615 (2013).
# 58. A Transcriptionally Distinct CXCL13þCD103þCD8þ T-cell Population Is Associated with B-cell Recruitment and Neoantigen Load in Human Cancer, cancer immunology research 2019
# 59. https://www.genecards.org/cgi-bin/carddisp.pl?gene=FCGR3A
# 60. https://www.genecards.org/cgi-bin/carddisp.pl?gene=FGFBP2
# 61. http://en.wikipedia.org/wiki/Common_gamma_chain
# 62. CYTOKINE CONTROL OF MEMORY T-CELL DEVELOPMENT AND SURVIVAL, Nature reviews immunology, 2003
# 63. The integration of T cell migration, differentiation and function, Nature reviews immunology, 2013
# 64. https://www.genecards.org/cgi-bin/carddisp.pl?gene=ANXA1
# 65. IL-17 and Th17 Cells, annual reviews immunology, 2009
# 66. Tumor-Infiltrating Regulatory T-cell Accumulation in the Tumor Microenvironment Is Mediated by IL33/ST2 Signaling, cancer immunology research 2020
# 67. ST2 as checkpoint target for colorectal cancer immunotherapy, JCI insight, 2019
# 68. Dissecting the multicellular ecosystem of metastatic melanoma by single-cell RNA-seq, Science, 2016
# 69. Tumor-Derived IL33 Promotes Tissue-Resident CD8 + T Cells and Is Required for Checkpoint Blockade Tumor Immunotherapy,  cancer immunology research 2020
# 70. Mutations Associated with Acquired Resistance to PD-1 Blockade in Melanoma, NEJM, 2016
# 71. S-100 protein: is it useful as a tumour marker in diagnostic immunocytochemistry?, 1980
# 72. interferon-Gamma at the Crossroads of Tumor immune Surveillance or evasion, frontieres immunology, 2018
# 73. Resistance to checkpoint blockade therapy through inactivation of antigen presentation, Nature communication, 2017

